api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider_test.go

```diff
@@ -95,7 +101,7 @@ func TestRead(t *testing.T) {
 	}
 	assert.Nil(t, err)
 
-	res, err := provider.Read("catalog1:v1", "components", nil)
+	res, err := provider.Read("catalog1:v1", ".components", nil)
```

Comment:
So, we always require the "." now? I think we should make it optional.

Reply:
I think we should still make simple cases straight-forward - like querying a property directly. Can we check if it's a valid json identifier first?

Reply:
updated. Add an input check for simple query strings.